### PR TITLE
Promote Learn & Documentation

### PR DESCRIPTION
**Pull request by wilzbach**
In my opinion "Learn" is more important than the community for newcomers a.k.a. I want to learn about the language, then I check how I can get involved. Same goes for "Documentation".

Also I found two minor typos:

1) Last item of an iteration should use "and"
2) "introduction tour to D" -> "Introduction to D" - does this sound better?

### CODE REVIEW

**Comment by aG0aep6G** [comment]
When choosing the order, I also had the lengths of the snippets in mind. If you put snippets with the same line count next to each other, it looks neater and minimizes white space. Of course, this can also be achieved by editing the text.

File: index.dd [hunk 1]
```diff
@@ -93,30 +93,21 @@ $(DIVC boxes,
...
         $(TOUR graduation-cap, Learn,
             $(P Quick start with the $(LINK2 http://tour.dlang.org,
-                introduction tour to D),
+                Introduction to D),
             	check out Ali Çehreli's great textbook
                 $(LINK2 http://ddili.org/ders/d.en/index.html,
                 <cite>Programming in D</cite>), ask questions in the
                 $(LINK2 https://forum.dlang.org/group/digitalmars.D.learn,
-                    Learn forum),
+                    Learn forum) and
```
Comment (aG0aep6G) [inline index.dd]: I like it better with just the comma. An "and" seems to say that you have to do all that. An "or" would seem to say that they're mutually exclusive. Just a comma is like an "and/or" between the elements. Or maybe I'm seeing a difference where there is none.

Anyway: Oxford comma.

**Comment by wilzbach** [comment]
> When choosing the order, I also had the lengths of the snippets in mind. 

Wow you did the redesign? In that case: Awesome work & thank you!!
I just found this initial mock where it has exactly the same ordering as now proposed: 

https://51e2a05fc1539b6822f2a4bd501de1331d7a3473-www.googledrive.com/host/0B7UtafxGD9vESlB3aFBxcjNPOXM/

> If you put snippets with the same line count next to each other, it looks neater and minimizes white space. Of course, this can also be achieved by editing the text.

Fair point - I would prefer to fix the text though.

While knowing the exact length of News is impossible due to it's changing nature we have two paragrahs there, but it doesn't seem difficult to fix & now we also provide a link to the great overview page:

image

There is also the [comparison](http://dlang.org/comparison.html), but I prefer if we merge that one into the overview soon.

> Anyway: Oxford comma.

Oh right :/

Btw from [a thread](http://forum.dlang.org/post/n53ps0$2j8f$1@digitalmars.com)  "[...] only HTML, JavaScript and Less (CSS). It's not a functioning site, it's only to show the design.".
How did we end up loosing Less?

**Comment by aG0aep6G** [comment]
> Fair point - I would prefer to fix the text though.

Sure thing. Also, it's not a hard rule that neighboring boxes have to be the same size, just something to consider.

Community section looks good.

Regarding the Learn section, do we even need getstarted.html anymore? Maybe we can integrate the information from there in the Learn box, and get rid of getstarted.html (i.e. redirect to the tour or to the home page, or keep it around as is but don't link there).

Another thought about the boxes: Maybe switch Documentation and Community, so that Learn and Documentation are visually more connected. Could also switch Packages and Contribute. Then the left column (News, Community, Contribute) would be more active, about getting involved, and the right column (Learn, Documentation, Packages) would be more passive, about using what we provide. Maybe I'm overthinking this.

> Btw from a thread "[...] only HTML, JavaScript and Less (CSS). It's not a functioning site, it's only to show the design.".
> How did we end up loosing Less?

It's all in that thread. As you see, Jacob Carlborg brought the thing forward. He [wanted to use vibe.d, Sass, CoffeeScript](http://forum.dlang.org/post/n595rd$h7g$1@digitalmars.com). But Andrei [wasn't a fan of that](http://forum.dlang.org/post/n59dc7$rsd$1@digitalmars.com). Jacob [didn't want to put up with that](http://forum.dlang.org/post/n5atii$2oj0$1@digitalmars.com). Since the two couldn't agree on a way forward, I went ahead and [pushed it without additional tools](http://forum.dlang.org/post/n5hhb0$h6g$1@digitalmars.com) (anonymous is me in that thread). So, Andrei is to blame, I guess :P

**Comment by wilzbach** [comment]
> Sure thing. Also, it's not a hard rule that neighboring boxes have to be the same size, just something to consider.

No it's a valid point and we should definitely consider it - this time I paid more attention when I reworded the Learn section ;-)

> Regarding the Learn section, do we even need getstarted.html anymore? Maybe we can integrate the information from there in the Learn box, and get rid of getstarted.html (i.e. redirect to the tour or to the home page, or keep it around as is but don't link there).

Yes I recently pushed the entire content of getting started to the Dlang Tour and we replace the menu link to the Tour, hence there is no other site pointing at this page. I added the remaining links for getstarted to Learn and killed getstarted!

I hope the reworded Learn section is a lot more useful now!

> Another thought about the boxes: Maybe switch Documentation and Community, so that Learn and Documentation are visually more connected. Could also switch Packages and Contribute. Then the left column (News, Community, Contribute) would be more active, about getting involved, and the right column (Learn, Documentation, Packages) would be more passive, about using what we provide. Maybe I'm overthinking this.

No you are not - I like the idea! Done!

> It's all in that thread. As you see, Jacob Carlborg brought the thing forward. He wanted to use vibe.d, Sass, CoffeeScript. But Andrei wasn't a fan of that. Jacob didn't want to put up with that. Since the two couldn't agree on a way forward, I went ahead and pushed it without additional tools (anonymous is me in that thread). So, Andrei is to blame, I guess :P

Arrgh :sob: 

There has been this [recent thread](http://forum.dlang.org/post/nhv9lu$2sd8$1@digitalmars.com) about using ddoc again for CSS which luckily didn't made it through.

**Comment by aG0aep6G** [comment]
Don't just delete getstarted.html. Whenever possible we try not to break links/bookmarks. Either set up a redirect in .htaccess, or leave the file up without linking to it.

File: index.dd [hunk 2]
```diff
@@ -93,28 +93,36 @@ $(DIVC boxes,
                 Adam D. Ruppe's weekly newsletter. $(SPANC twid)
             )
         )
+        $(TOUR graduation-cap, Learn,
+            $(P Quick start with $(LINK2 http://tour.dlang.org,
+                Introduction to D), explore
```
Comment (aG0aep6G) [inline index.dd]: Is the name "Introduction to D" used anywhere else? The site itself seems to use "The Dlang Tour". How about "Take the [tour](http://tour.dlang.org),"?

File: index.dd [hunk 3]
```diff
@@ -93,28 +93,36 @@ $(DIVC boxes,
...
+                $(LINK2 $(ROOT_DIR)overview.html, major features) in D,
+                read coming from $(LINK2 $(ROOT_DIR)ctod.html, C) or
```
Comment (aG0aep6G) [inline index.dd]: "read coming from C or C++" sounds off to me. I don't have a better suggestion, though.

File: index.dd [hunk 4]
```diff
@@ -93,28 +93,36 @@ $(DIVC boxes,
...
+                and ask questions in the
+                $(LINK2 https://forum.dlang.org/group/digitalmars.D.learn,
+                    Learn forum).
+            )
+
+            $(P For a deeper dive into D
+            	check out $(LINK2 https://wiki.dlang.org/Books, books) about D, e.g.
+            	Ali Çehreli's
+                $(LINK2 http://ddili.org/ders/d.en/index.html,
+                <cite>Programming in D</cite>) for beginners.
```
Comment (aG0aep6G) [inline index.dd]: Is it correct to label Ali's book as "for beginners"? Personally, I'd put "(it's free)" or "(pay what you want)" there instead.

File: index.dd (hunk 3 above)
Comment (wilzbach) [inline index.dd]: just some ideas:

- learn D with C, or C++ background
- compare D with C, or C++
- see Programming in D for C, or C++ programmers (that is probably too long)

Comment (aG0aep6G) [inline index.dd]: > compare D with C, or C++

I like that one. I'd drop the comma.

File: index.dd (hunk 4 above)
Comment (wilzbach) [inline index.dd]: "Programming in D is an extensive tutorial by Ali Çehreli and a great starting point for __absolute beginners__. It doesn't assume that you know any other programming languages, and it also teaches you about some fundamental concepts of programming."

From http://dlang.org/getstarted.html

> I'd put "(it's free)" 

Would be fine too.

> or "(pay what you want)" there instead.

Maybe too long or confusing for a start page?

Comment (CyberShadow) [inline index.dd]: Yes, Ali's book is unusual in that it is accessible even to people new to programming. Contrast e.g. with TDPL which assumes knowledge of other languages and general concepts.

**Comment by wilzbach** [comment]
> Don't just delete getstarted.html. Whenever possible we try not to break links/bookmarks. Either set up a redirect in .htaccess, or leave the file up without linking to it.

Makes sense - turns out there's a "Getting started" page in the [wiki](http://wiki.dlang.org/Getting_Started) too - I made the redirect over there.
If at some point in the future we decide to rewrite that page, we have redirects from the wiki ;-)

File: index.dd (hunk 4 above)
Comment (aG0aep6G) [inline index.dd]: > "Programming in D is an extensive tutorial by Ali Çehreli and a great starting point for absolute beginners. It doesn't assume that you know any other programming languages, and it also teaches you about some fundamental concepts of programming."

Yeah, I think I wrote that. I didn't mean to say that it's only for newbies, but that it's suitable for newbies. But maybe it is actually aimed mostly at newbies.

I'd go without "for beginners", but I don't mind if it stays.

**Comment by aG0aep6G** [comment]
I'm not 100% about the redirect target (alternatives would be the tour or the home page), but it's ok.

Looks good to me.

File: index.dd (hunk 4 above)
Comment (wilzbach) [inline index.dd]: > I'd go without "for beginners", but I don't mind if it stays.

For me both ways are also ok. How about we remove it and use the freed space to say "about D, among others Ali Çehreli's _Programming in D_"?

Comment (aG0aep6G) [inline index.dd]: > How about we remove it and use the freed space to say "... books about D, among others Ali Çehreli's Programming in D"?

Looks good to me.

**Comment by wilzbach** [comment]
> I'm not 100% about the redirect target (alternatives would be the tour or the home page), but it's ok.

Thought about both.

- We will change the URLs naming scheme for the tour soon https://github.com/stonemaster/dlang-tour/issues/137
- Redirecting to the frontpage is a bit weird if there's a getting started link out there, but would work for me too ;-)

> Looks good to me.

Should I squash all commits?

**Comment by CyberShadow** [comment]
I think the "Review Queue" link should go too. That page is pretty stale.

**Comment by wilzbach** [comment]
> I think the "Review Queue" link should go too. That page is pretty stale.

Absolutely - that's #1333.

**Comment by wilzbach** [comment]
Rebased to latest master ;-)

**Comment by CyberShadow** [comment]
Something has gone wrong in the generated HTML, the page is broken now.

File: index.dd [hunk 5]
```diff
@@ -93,29 +93,36 @@ $(DIVC boxes,
                 Adam D. Ruppe's weekly newsletter. $(SPANC twid)
             )
         )
+        $(TOUR graduation-cap, Learn,
+            $(P Take the $(LINK2 http://tour.dlang.org,
+                Tour), explore
+                $(LINK2 $(ROOT_DIR)overview.html, major features) in D,
+                compare D with $(LINK2 $(ROOT_DIR)ctod.html, C) or
+                $(LINK2 $(ROOT_DIR)cpptod.html, C++),
```
Comment (CyberShadow) [inline index.dd]: Best not. Language comparisons are often seen as controversial and attract flame wars.

These pages are meant to provide an intro to D using concepts from other languages, not as a language comparison.

Comment (wilzbach) [inline index.dd]: > Best not. Language comparisons are often seen as controversial and attract flame wars.

how about "intro with C or C++ background" - or should I remove it entirely?

**Comment by wilzbach** [comment]
> Something has gone wrong in the generated HTML, the page is broken now.

Argh - should be fixed now.

**Comment by wilzbach** [comment]
> how about "intro with C or C++ background" - or should I remove it entirely?

I renamed it to "quick start with C or C++ background" - is this clear enough to avoid flame wars?

**Comment by CyberShadow** [comment]
Perfect!

**Pull request merged**